statsmodels/regression/tests/test_mixed_glm.py

```diff
@@ -0,0 +1,350 @@
+    par = MixedGLMParams.from_components(fe_params=np.asarray([0.9920529, 0.9011868]),
+                                         cov_re=np.asarray([[0.59920**2]]))
+    logLik = model.loglike(par)
+    np.testing.assert_allclose(logLik, -61.3776, rtol=1e-6)
+
+# Can't check Gaussian since it uses the concentrated log-likelihood
+
+import statsmodels.genmod.families.links as links
+
+# Family instances
+class CheckFamily(object):
+
+    family = sm.families.Family
+    test_links = []
+
+    def test_links_derivs(self):
+        for lnk in self.test_links:
+            try:
+                self.like_hess_grad(self.family, lnk)
+            except Exception as e:
+                e.args += ("Link type: " + str(lnk),)
+                raise
+
+    def like_hess_grad(self, fam, lnk):
+        """
+        Test the function, gradient and Hessian of the joint
+        log-likelihood of the random effects and the data.
+        """
+
+        fa = fam(link=lnk)
+
+        np.random.seed(423)
+        endog, exog, groups = gen_mixed(fa.__class__.__name__, 10)
+
+        model = MixedGLM(endog, exog, groups=groups, family=fa)
+
+        # Set some parameters
+        fe_params = np.random.normal(size=2)
+        cov_re = np.abs(np.random.normal(size=(1,1)))
+        scale = 1.0
+
+        self.like_hess_grad_helper(model, fe_params, cov_re, scale)
+
+    def test_links_derivs2(self):
+        for lnk in self.test_links:
+            try:
+                self.like_hess_grad2(self.family, lnk)
+            except Exception as e:
+                e.args += ("Link type: " + str(lnk),)
+                raise
+
+    def like_hess_grad2(self, fam, lnk):
+        """
+        Test the function, gradient and Hessian of the joint
+        log-likelihood of the random effects and the data.
+        """
+
+        fa = fam(link=lnk)
+
+        np.random.seed(423)
+        endog, exog, groups, rs = gen_mixed2(fa.__class__.__name__, 10)
+        exog_re = np.vstack((np.ones(len(groups)),rs)).T
+
+        model = MixedGLM(endog, exog, groups=groups, exog_re=exog_re, family=fa)
+
+        # Set some parameters
+        fe_params = np.random.normal(size=2)
+        cov_re = np.abs(np.random.normal(size=(2,2)))
+        cov_re = np.dot(cov_re.T,cov_re)
+        scale = 1.0
+
+        self.like_hess_grad_helper(model, fe_params, cov_re, scale)
+
+
+    def like_hess_grad_helper(self, model, fe_params, cov_re, scale):
+
+        # Get the log-likelihood function, and its gradient and Hessian.
+        fungradhess = model._gen_joint_like_grad_hess(fe_params, cov_re, scale)
+        fun1 = lambda x : fungradhess(x)[0]
+        grad = lambda x : fungradhess(x)[1]
+        hess = lambda x : fungradhess(x)[2]
+
+        for k in range(10):
+
+            RTOL = 1e-2
+            ATOL = 1e-2
+
+            # An arbitrary state for the random effects
+            ref = np.random.normal(size=(model.n_groups, model.k_re)).ravel()
```

Comment:
Based on the first test error: fun is nan
are these parameters all valid?  My impression (just getting started so I haven't figured it out yet) is that the unpacked params are not arbitrary.